### ISSUE TITLE
`with_` syntax is long

### ISSUE THREAD

**Issue by kellpossible** [skubalj/build_html#3]
The `with_` syntax for adding an element is rather long, assuming this gets used to build some rather large HTML structures, it seems like a lot of unnecessary typing, perhaps you could consider dropping the `with_`?

**Comment by skubalj** [comment]
Thanks for commenting!

I'm not a huge fan of the idea of shortening the names. Personally, I like having the `with_` because it makes both the self-consuming (`with`) and self-modifying (`add`) APIs more parallel, without showing direct preference for either approach. From a UX perspective, I think that making one version of the API shorter would cement it as the "default" choice, when each version is correct in different circumstances. For example, if I was aggregating an iterator with a `fold`, I might prefer to use the `with` methods, whereas if I was using a `for` loop, I might prefer to use the `add` methods.

I also like having the verb because it makes reading the code feel a little more like natural language, similar to using `is_empty()` rather than just `empty()` to check whether a slice has any elements in it. Obviously, this is all completely semantic and philosophical, so we're free to disagree on whether that's worth a few additional keystrokes! :slightly_smiling_face: 

In general, I don't know that I would prefer build_html for very large constructs. If I was writing lots of HTML, say for a more complicated page, I might try to extract the parts that don't change and write them as their own HTML file that can be formatted and linted by my editor. I would use build_html primarily to fill in the pieces that I need to generate programatically. Then, I could embed the HTML file with the `include_str!()` macro and patch in the extra HTML pieces with `format!()`. That's not to say that you *can't* use build_html to make something bigger, but once things start to get complicated, I find that it's nice to have my editor tooling help me work in actual HTML.

If you do want to generate a large page completely from within Rust, (and I acknowledge that there are definitely a few reasons that you might want to!) my first thought might be to look at macros. Hypothetically, a higher-level macro-based API could rely on build_html under the hood to make something even more ergonomic than just shortening the method names. Those macros could be split out into an ecosystem of crates providing different APIs for different use cases. For example, maybe one API could be used to help simplify generating react components, while another helps apply styles to your HTML. 

As I said, most of this is semantic and personal preference, so there's certainly room to disagree.

**Comment by kellpossible** [comment]
Thanks! I respect your reasoning. I used https://docs.rs/html-builder/latest/html_builder/ in another project, which generates the html for a page entirely in Rust. I enjoyed using it, and wanted to explore the space further (thus finding your library) before starting on the next one. For me, the main motivation to use this approach is to have code that is in a single well established programming language, that is simple (ideally no extra dependencies), easy to understand and easy to refactor using standard tools if required. In my opinion, the macro based approaches lose on this, I would almost rather use one of the template language implementations, at least they are approaching some kind of well established standard rather than some kind of bespoke DSL.
